src/parser/parser.cpp

```diff
@@ -157,7 +156,7 @@ void Parser::ParseQuery(const string &query) {
 		if (options.extensions) {
 			for (auto &ext : *options.extensions) {
 				D_ASSERT(ext.parse_function);
-				auto result = ext.parse_function(ext.parser_info.get(), query);
+				auto result = ext.parse_function(ext.parser_info.get(), query, parser_error);
```

Comment:
This is the change.. I didn't realize I couldn't open a PR towards another [PR](https://github.com/duckdb/duckdb/pull/7745). I meant for this to go in after #7745 goes in... Since on my side I'm coding against all of these changes.. Sorry about the confusion.

Reply:
Isn't that already returned in the result?
```c++
struct ParserExtensionParseResult {
	ParserExtensionParseResult() : type(ParserExtensionResultType::DISPLAY_ORIGINAL_ERROR) {
	}
	ParserExtensionParseResult(string error_p)
	    : type(ParserExtensionResultType::DISPLAY_EXTENSION_ERROR), error(std::move(error_p)) {
	}
	ParserExtensionParseResult(unique_ptr<ParserExtensionParseData> parse_data_p)
	    : type(ParserExtensionResultType::PARSE_SUCCESSFUL), parse_data(std::move(parse_data_p)) {
	}

	//! Whether or not parsing was successful
	ParserExtensionResultType type;
	//! The parse data (if successful)
	unique_ptr<ParserExtensionParseData> parse_data;
	//! The error message (if unsuccessful)
	string error;
};
```

Reply:
True, thanks! Closing this PR.

Reply:
Yeah... so for now, I'm just catching and returning `duckdb::ParserExtensionParseResult();` (as @Tishj suggested) because we don't actually throw our own parser errors yet. But in the future, I will need a way to distinguish and decide which error to throw (ours vs. DDB's) so we may need to make more changes at that time.